Fix #16095 Leading space not shown in a CHAR column

petrduda opened the pull request:
### Description

Add "white-space: pre-wrap;" css styling to value cells of non-numeric columns other than date types, provided that no-wrapping is not enabled by transform functions.

Fixes #16095

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

petrduda committed:
Fix #16095 - Leading space not shown in a CHAR column

Signed-off-by: Petr Duda <petrduda@seznam.cz>

williamdes reviewed (approved):
It looks very good !
Could you add more issue related test cases to dataProviderForTestGetDataCellForNonNumericColumns ?

petrduda committed:
Fix #16095 - Add test cases for displaying results in non-numeric columns

Signed-off-by: Petr Duda <petrduda@seznam.cz>

petrduda commented:
I added two test cases for non numeric columns: no-wrapping enabled by transform function and no-wrapping for datetime field type.

I think it is not necessary to have another test for a string with leading spaces, because they are not trimmed anywhere on the way. But perhaps the replacement of two whitespaces with ' &nbsp;' in Core->mimeDefaultFunction() is not necessary?

In the transformation_plugin->function_nowrap() condition, I corrected a small error which made the condition always evaluate as false (function_exists vs. method_exists).

williamdes reviewed (approved):
This looks very good !
> In the transformation_plugin->function_nowrap() condition, I corrected a small error which made the condition always evaluate as false (function_exists vs. method_exists).

:100: 

> I think it is not necessary to have another test for a string with leading spaces, because they are not trimmed anywhere on the way. But perhaps the replacement of two whitespaces with ' &nbsp;' in Core->mimeDefaultFunction() is not necessary?

Maybe @mauriciofauth agrees ?
